Type 'unknown' is not assignable to type 'RouterReducerState<SerializedRouterStateSnapshot>

`Error: src/app/app.module.ts:31:26 - error TS2322: Type '<T extends BaseRouterStoreState = SerializedRouterStateSnapshot>(state: RouterReducerState<T> | undefined, action: Action) => RouterReducerState<...>' is not assignable to type 'ActionReducer<unknown, Action>'.
  Types of parameters 'state' and 'state' are incompatible.
    Type 'unknown' is not assignable to type 'RouterReducerState<SerializedRouterStateSnapshot> | undefined'.
      Type 'unknown' is not assignable to type 'RouterReducerState<SerializedRouterStateSnapshot>'.

31     StoreModule.forRoot({router: routerReducer, count: counterReducer, movies: movieReducer, }),
                            ~~~~~~



`



`
    StoreModule.forRoot({router: routerReducer, count: counterReducer, movies: movieReducer, }),`

An error occurred when routerReducer and counterReducer are referenced together

No error will be reported if routerReducer is used alone

--- a/modules/router-store/src/reducer.ts
+++ b/modules/router-store/src/reducer.ts
@@ -16,22 +16,20 @@ export type RouterReducerState<
 };
 
 export function routerReducer<
-  T extends BaseRouterStoreState = SerializedRouterStateSnapshot
->(
-  state: RouterReducerState<T> | undefined,
-  action: Action
-): RouterReducerState<T> {
+  RouterState extends BaseRouterStoreState = SerializedRouterStateSnapshot,
+  Result = RouterReducerState<RouterState>
+>(state: Result | undefined, action: Action): Result {
   // Allow compilation with strictFunctionTypes - ref: #1344
-  const routerAction = action as RouterAction<any, T>;
+  const routerAction = action as RouterAction<any, RouterState>;
   switch (routerAction.type) {
     case ROUTER_NAVIGATION:
     case ROUTER_ERROR:
     case ROUTER_CANCEL:
-      return {
+      return ({
         state: routerAction.payload.routerState,
         navigationId: routerAction.payload.event.id,
-      };
+      } as unknown) as Result;
     default:
-      return state as RouterReducerState<T>;
+      return state as Result;
   }
 }

